community-tutorials/workflow-automation-with-n8n-and-netcup-webhosting/01-en.md

```diff
@@ -0,0 +1,128 @@
+
+# Step 2 - Setup a domain
+In this step, we are going to create a new subdomain for our n8n instance. On Netcup each Webhosting gets a free `hostingXXXXXX.xxxxx.netcup.net` domain. In this tutorial we will use this one to host n8n. If you want to setup a custom domain, look at the [official documentation](https://www.netcup-wiki.de/wiki/Domains_CCP).
+
+## Step 2.1 - Create a new subdomain
+To add a new subdomain, click **New Subdomain** at the top left corner. This will guide you to a new page, where you can select the parental domain and the document root of your subdomain.
+
+In this case I will with our inclusive domain and `/httpdocs/n8n/public` as the document root.
+> Please make sure to add `/public` at the end of the document root. Otherwise, you will have problems with NodeJS.
+
+![Create a new subdomain](./images/step2-new-subdomain.png)
+
+## Step 2.2 - Setup HTTPS with Let's Encrypt
+After the domain is created, we need to issue a Let's Encrypt certificate for it, so we can access our domain over HTTPS.
+
+To do this, select the domain at the overview and click the Let's Encrypt button.
+
+In the following window you now must enter your Email address. After that you can click **Install** to issue a new certificate.
+
+>Please note that Netcup will renew this certificate for you automatically.
+
+![Issue a Let's Encrypt Certificate](./images/step2-lets-encrypt.png)
+
+# Step 3 - Add the code
+Now that we have our domain ready and set up, let's concentrate on the coding part.
+
+So, let's jump over to our `/httpdocs/n8n` directory (this time without `/public` 😉), you can do this by clicking **Files** on the left or **File Manager** at the domain config and add the following two files:
+
+`package.json`
+```json
+{
+  "name": "n8n",
+  "version": "1.0.0",
+  "main": "index.js",
+  "license": "MIT",
+  "dependencies": {
+    "n8n": "*",
+    "sqlite3": "*"
+  }
+}
+```
+
+and
+
+`.npmrc`
+```
+scripts-prepend-node-path=true
+```
+
+In the first file we define the version of n8n and sqlite3 (a dependency of n8n), we want to install (in this case the latest: `"*"`). The second one just ensures that everything is fine on our shared webhosting.
+
+# Step 4 - Start the app
+Now that everything is in place, it is time to setup the final NodeJS configuration.
+
+To do this, go to **Websites & Domains** on the left side bar and select your domain. Then click **Node.js** and you will see a new page. Here click on the **Enable Node.js** button to active Node.
+
+Now that Node is running you can specify a Node.js Version. I recommend using the latest, but you must try which one works with n8n. Currently (November 2021) Node.js version 14 is working well.
+
+## Step 4.1 - Install NPM dependecies
+Now that Node is up and running, it's time to install all dependencies. You can do this by clicking the **NPM install** button at the top menu.
+
+If everything works fine, it should exit with some warnings, but we can ignore these.
```

Comment:
Unfortunately I got some errors while installing the dependencies:

- /bin/sh: cmake: command not found
- make: *** [/var/www/vhosts/hosting164325.a2e22.netcup.net/httpdocs/n8n/node_modules/cpu-features/deps/cpu_features/build/Makefile] Error 127
- gyp ERR! build error
- gyp ERR! stack Error: `make` failed with exit code: 2
- gyp ERR! stack at ChildProcess.onExit (/opt/plesk/node/14/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
- gyp ERR! stack at ChildProcess.emit (events.js:400:28)
- gyp ERR! stack at Process.ChildProcess._handle.onexit (internal/child_process.js:282:12)
- gyp ERR! System Linux 4.9.0-16-amd64
- gyp ERR! command "/opt/plesk/node/14/bin/node" "/opt/plesk/node/14/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
- gyp ERR! cwd /var/www/vhosts/hosting164325.a2e22.netcup.net/httpdocs/n8n/node_modules/cpu-features
- gyp ERR! node -v v14.18.1
- gyp ERR! node-gyp -v v5.1.0
- gyp ERR! not ok

Reply:
It should work anyways. However i forgot to mention that the application script path needs to be changed (sorry for that :cry:). I updated the doc so everything should work fine now (please ignore this error, it has to do with plesk).